src/iop/globaltonemap.c

```diff
@@ -522,12 +523,8 @@ int process_cl(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, cl_m
     if(err != CL_SUCCESS) goto error;
     // and apply it to output buffer after logscale
     err = dt_bilateral_slice_to_output_cl(b, dev_in, dev_out, d->detail);
-    if(err != CL_SUCCESS) goto error;
-    dt_bilateral_free_cl(b);
   }
 
-  return CL_SUCCESS;
```

Comment:
Even if correct, the only issue I have with that is that on success we go through the `error:` block. So either keep the last `return CL_SUCCESS` plus memory release or rename the `error:` label to `finally:` ?